### PR TITLE
docs(plan): parallel batch file-level conflict detection

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Implementation plan for issue #324: before dispatching a parallel batch, the orchestrator detects file-level conflicts by extracting declared file sets from implementation plan documents and cross-checking for overlaps between items in the same proposed batch.

**Approach**: Extend `workflow-batch-plan.sh` with two new helpers — `extract_file_set` (parses the implementation plan's explicit file list via fenced code block or bullet list under a matching heading) and `detect_file_conflicts` (pairwise overlap check with priority-based serialization per BR-4/BR-5). Add a new "Same-batch file-level conflict detection" subsection to Protocol 90 Step 3, placed after the tool-fix ordering hazard and before the Codex fallback paragraph.

**Complexity**: M — two files changed (one shell script, one protocol document), but the extraction parser requires careful handling of heading variants and edge cases.

**Key risks**: Heading-normalization misses for unusual plan formats; priority tiebreaker logic for equal-priority items; BR-8 ordering discipline (tool-fix check runs first, conflict detection only sees the remaining items).

## Links

- Spec: `docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/1_324-parallel-batch-file-conflict-detection_specs.md`
- Plan: `docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md`
- Smoke test runbook: `docs/testing/workflow/324-parallel-batch-file-conflict-detection.smoke-test.md`

### CODE REVIEW

File: docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+## Implementation Order
+
+1. Read `scripts/development-workflow/workflow-batch-plan.sh` and
+   `scripts/development-workflow/workflow-lib.sh` in full to understand existing helpers
+   (`print_kv`, `batch_hint_for_action`, `parallel_safe_for_action`, `classify_tool_fix`,
+   `extract_github_issue_number`, `cd_workflow_repo_root`).
+
+2. Add `extract_file_set` helper function to `workflow-batch-plan.sh`. Place it after
+   `classify_tool_fix` and before `cd_workflow_repo_root`. Implement format detection for:
+   - Fenced code block under a heading matching `/files\s+(to\s+be?\s+)?modified/i`
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md]: Fixed in commit `fec82a4`. The heading regex has been updated from `be?` to nested groups `(to\s+(be\s+)?)?` so it correctly matches `files modified`, `files to modified`, and `files to be modified` — aligning with the Layer-by-Layer spec and edge-case table.

File: docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+while i < len(lines):
+    if heading_re.search(lines[i]):
+        i += 1
+        # Try fenced code block
+        if i < len(lines) and lines[i].startswith('```'):
+            i += 1
+            while i < len(lines) and not lines[i].startswith('```'):
+                p = lines[i].strip()
+                if p:
+                    paths.append(p.lstrip('/').replace('\\\\', '/'))
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md]: Fixed in commit `fec82a4`. The `replace('\\\\', '/')` calls on lines 292 and 299 have been corrected to `replace('\\', '/')`. In a single-quoted heredoc (`<<'PYEOF'`), bash passes the source verbatim to Python; `'\\\\'\ ` (4 chars in the source) is two backslashes in Python, which would replace double-backslash sequences only. The fix `'\\'\ ` (2 chars) is one backslash in Python, correctly normalizing single-backslash paths like `docs\\foo\\bar.sh` to `docs/foo/bar.sh` as specified in the edge-case table.

File: docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+                    paths.append(p.lstrip('/').replace('\\', '/'))
+                i += 1
+        else:
+            # Bullet list
+            while i < len(lines) and (lines[i].startswith('- ') or lines[i].startswith('* ')):
+                p = lines[i][2:].strip()
+                if p:
+                    paths.append(p.lstrip('/').replace('\\', '/'))
+                i += 1
+    else:
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md:305]: Fixed in commit `6cdf0ff`. Added a blank-line-skip loop (`while i < len(lines) and not lines[i].strip(): i += 1`) immediately after the heading match so the parser correctly handles blank lines between the heading and the fenced code block or bullet list. Also added a new edge-case table row documenting this case.

File: docs/testing/workflow/324-parallel-batch-file-conflict-detection.smoke-test.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+**Expected result**:
+- Output contains `FILE_SET=` with both paths listed
+- Paths are comma-separated and sorted
+- No `FILE_SET=unknown` in the output
+
+**Maps to**: AC-1, AC-2 (known file set correctly extracted), BR-2
+
+---
+
+### Step 2: AC-3 — `extract_file_set` with bullet-list format
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/testing/workflow/324-parallel-batch-file-conflict-detection.smoke-test.md]: Fixed in commit `6cdf0ff`. Step 2's title and fixture directory/plan title have been updated from `AC-3` to `AC-1` (and from `ac3-bullet-list` to `ac1-bullet-list`), correctly reflecting that this step tests known-file-set extraction via bullet-list format, which maps to AC-1. Step 3 remains the AC-3 test (no explicit file list).

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean
**Cycles:** 2 / 10
**Platforms:** devin

| # | Platform | File | Line | Description | Status | Resolved In |
|---|----------|------|------|-------------|--------|-------------|
| 1 | devin | `docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md` | 323 | Heading regex \`be?\` contradicts Layer-by-Layer spec and edge-case table... | Resolved | \`fec82a4\` |
| 2 | devin | `docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md` | 292 | Code sample \`replace('\\\\\\\\', '/')\` replaces double backslashes, not single... | Resolved | \`fec82a4\` |
| 3 | devin | `docs/specs/developments/20260427222126_324-parallel-batch-file-conflict-detection/2_324-parallel-batch-file-conflict-detection_implementation-plan.md` | 301 | Code sample and edge-case table fail to handle blank lines between heading and content block | Resolved | \`6cdf0ff\` |
| 4 | devin | `docs/testing/workflow/324-parallel-batch-file-conflict-detection.smoke-test.md` | 75 | Smoke test Step 2 title mislabeled as AC-3 instead of AC-1 | Resolved | \`6cdf0ff\` |

**Resolved:** 4 / 4 findings
